### PR TITLE
Interpret d_discontinuity numbers as zeroth order discontinuities

### PR DESCRIPTION
**Pull request by ChrisRackauckas**

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
You deserve a cookie.

File: src/solve.jl [hunk 1]
```diff
@@ -462,10 +462,10 @@ function OrdinaryDiffEq.tstop_saveat_disc_handling(tstops, saveat, d_discontinui
   end
 
   for d in d_discontinuities
-    tdir_t = tdir * d.t
-
-    if tdir_t0 < tdir_t < tdir_tf && d.order ≤ alg_maximum_order + 1
-      push!(d_discontinuities_internal, Discontinuity{tType}(tdir_t, d.order))
+    tdir_t = d isa Discontinuity ? tdir * d.t : tdir * d
+    order = d isa Discontinuity ? d.order : 0
+    if tdir_t0 < tdir_t < tdir_tf && 0 ≤ alg_maximum_order + 1
```
Comment (devmotion) [inline src/solve.jl]:
```suggestion
    if tdir_t0 < tdir_t < tdir_tf && order ≤ alg_maximum_order + 1
```

**Comment by ChrisRackauckas** [comment]
Thanks, I was wondering why tests were failing locally.

**Comment by ChrisRackauckas** [comment]
Passed locally, and CI will be blocked for days due to tagbot so merging.

**Pull request merged**